### ISSUE TITLE
Shortcoming for validation of PredefinedType setting in entity facet

### ISSUE THREAD

**Issue by MatthiasWeise** [buildingSMART/IDS#214]
I was wondering if correct use of the predefinedType is checked by http://www.xbim.it/ids. 
Obviously not, because current agreement with USERDEFINED does enable to check misspelling of predefinedType.

**Comment by CBenghi** [comment]
Hi @MatthiasWeise,
Constraining the PredefinedType is something I originally added, but have since relaxed.
Indeed the code infrastructure to enforce the closed list is still in the tool, but the agreement in the group is that we must allow for the selection of user defined values.
What I can easily do is to add a warning to the log, that notifies of non-standard strings, if that helps.
Would that be useful?

**Comment by MatthiasWeise** [comment]
Yes, a warning would be good to avoid misspelling of predefinedType.

**Comment by CBenghi** [comment]
However that would prevent us from selecting specific parts of the models that have a custom predefinedType, which does not sound right.
We do need a test case that deals with some predefined type configuration, if we don't have one yet. I'll have to check.

**Comment by MatthiasWeise** [comment]
I was more thinking that we need both, USERDEFINED to be unspecific about what user defined type should be selected and to use any other value not being in the list of predefinedTypeEnums for a specific used defined type.

**Comment by andyward** [comment]
Agreed, checking for any custom/USERDEFINED predefined type feels useful. This shouldn't preclude implicitly checking the ObjectType.

1. Standard case: _All Doors defined as TrapDoor_ (PredefinedType = TRAPDOOR)

Entity Facet:
- IfcType: IFCDOOR
- PredefinedType: TRAPDOOR

2. User defined scenario _All Doors defined as 'SecurityDoor'_ (ObjectType = 'SecurityDoor' [and PredefinedType = USERDEFINED])

Entity Facet:
- IfcType: IFCDOOR
- PredefinedType: SecurityDoor

3. Edge case _All Doors with **any** User-defined type (PredefinedType = USERDEFINED)_

Entity Facet:
- IfcType: IFCDOOR
- PredefinedType: USERDEFINED


In particular, a use case might be: "All elements must NOT have a PredefinedType of USERDEFINED". That kind of test might help with #116, as presumably you could select applicable items with IfcFlowTerminal of Predefined Type 'DIFFUSER,GRILL' etc if you can exclude USERDEFINED and NOTDEFINED . This assumes implementations fall back to the Type to check the PDT - i.e [IfcAirTerminal]Type's PDT to filter the Ifc2x3 FlowTerminals.

It would also let IDS police the (mis)usage we sometimes see of IfcObject.ObjectType being defined when PredefinedType is NOTDEFINED - by combining an Entity requirement and an Attribute requirement.
